sympy/solvers/solvers.py

```diff
@@ -228,7 +228,7 @@ def checksol(f, symbol, sol=None, **flags):
                     return False
                 # there are free symbols -- simple expansion might work
                 _, val = val.as_content_primitive()
-                val = expand_mul(expand_multinomial(val))
+                val = simplify(expand_mul(expand_multinomial(val)))
```

Comment:
Thanks @smichr .
Then [in this](https://github.com/sympy/sympy/pull/11339/files#diff-e2a46e260ac6a40415353a416f21fd4fR240) place `simplify` for `val` will be good.